src/libs/ReportUtils.ts

```diff
@@ -3459,14 +3473,78 @@ function shouldHideReport(report: OnyxEntry<Report>, currentReportId: string): b
     return parentReport?.reportID !== report?.reportID && !isChildReportHasComment;
 }
 
+/**
+ *  Checks if there are any violations belonging to the transaction in the transactionsViolations Onyx object
+ *  then checks that the violation is of the proper type
+ */
+function transactionHasViolation(transactionID: string, transactionViolations?: TransactionViolations): boolean {
```

Comment:
Is it possible to make `transactionViolations` here and in `transactionThreadHasViolations` mandatory?